### DISCUSSION TITLE
New option MeterCompensation in H3 Master Firmware 2.0

### DISCUSSION THREAD

**Discussion by xrad** [nathanmarlor/foxess_modbus#645] [Q&A]
Hello there,

Fox has started distributing the Master Firmware 2.0 for the H3. With an installer account it offers a new option called MeterCompensation which has been observed to quite drastically reduce the grid consumtion when there is enough battery engery during the night. Unfortunately the option is not present in the system menu or for end-users. 

Does anyone here know more about this, specifically how it might work? Obviously it would be perfect if  a modbus register could be identified for this.

**Comment by maddin4711** [comment]
The new parameter “MeterCompensation” can be used to reduce the permanent grid consumption (typically 30 W) during battery operation (discharging and charging).

For example my inverter:
With the parameter “MeterCompensation = -18W”, an average of 25W is now fed into the grid during battery operation instead of drawing 30W from the grid as before. 

Unfortunately, the parameter can only be changed via the Agent or Installer account. Therefore, an option via Modbus would be desirable.

**Comment by xrad** [comment]
I have read about this result (I'm still on 1.98) but my question was more about the actual workings of the parameter. Like why exactly is it named "MeterCompensation", and why does it result in a the described behavior? Guessing from its name it is some kind of adjustment for the Smartmeter readings, but how is it applied exactly? How does it relate to the phases? And could it have an effect when there is no battery usage?

I'm aware that Fox should answer this but t he manual hasn't been updated and I don't think there's a changelog for 2.0 yet, but I was wondering if someone around here knows more about this.

**Reply by maddin4711** [reply]
I think “MeterCompensation” refers to the measured value of the DTSU666 meter, which the inverter uses to measure the feed-in and grid consumption. (In Home Assistant “Grid CT”; sum of R/S/T)

In my opinion, the “MeterCompensation” parameter shifts the threshold to which the inverter regulates in order to reach 0 watts. 
But only Fox knows the exact function.

Here is the comparison Master 1.98 to 2.00 with MeterCompensation = -18W
![Grid_CT_ohne_MeterCompensation](https://github.com/user-attachments/assets/1e72518b-d275-44a8-a845-121007572781)
![Grid_CT_MeterCompensation -18W](https://github.com/user-attachments/assets/354ee559-ff60-42ab-ab34-4a7eb7400a12)

**Comment by horstvogel** [comment]
Are there any changelogs available for fox ess Firmware

**Reply by FozzieUK** [reply]
Not as yet, we've asked many times - Fox have agreed to generate something but on last months update they hadn't resolved how/who would write them for public consumption, it'll still be on the list on the next call with them. 😞

**Reply by xrad** [reply]
Thanks for your efforts! Incidentally, do you know if they plan to update the product manuals? It would be nice to have that new option documented.

**Reply by FozzieUK** [reply]
It's a personal view but I think this is one of their biggest weaknesses - they do update warranty and specifications regularly but product manuals don't seem to change unless there is a regulatory reason to do it. 
I have a (sad) habit of reading through the latest manuals of the latest product as they often incorporate any significant changes there.

**Comment by horstvogel** [comment]
I think I have a very, very old version. My electrician installed the inverter in September last year. Have there been so many newer versions released since then? Unfortunately, since there is no changelog, you cannot find out. Could he then also do the update via Fox Cloud? Or is not possible for a normal electrician.He has a very small business, he has "installer" access for the inverter.
thanks


H3-6.0-E Inverter

![image](https://github.com/user-attachments/assets/645bede3-d831-4beb-99d2-e2335f2d610c)

**Reply by xrad** [reply]
You should login to the fox cloud, open a ticket there and ask for an update to the latest software version. Remember to tell them the ID of your inverter. Usually they do this rather quickly, 12-24h max. Don't know about weekends though. The electrician should also be able to do an update, but I would recommend going through Fox.

**Comment by derhoch** [comment]
Issue #853 links to a google drive with a relatively recent protocol definition (2025-01-15) that lists register 49248 as "MeterCompensation" for the H3 inverter.  I have the latest firmware and my technician set the value via cloud interface. It's working, but I cannot read the modbus register. Maybe it's wrongly listed as RW but is WO?

**Reply by trevevs** [reply]
this is illegal reg for KH10 - anyone know the register for KH10 grid comp?  Some places say only available to installers / support. Is that the case?  I sent a ticket to fox to set mine to -50 but  Idont believe they did it even though they said they did.  Wanted to check.  Im still using grid power and most days losing my zero hero $1 bones due to using power in peak.

**Comment by derhoch** [comment]
Well, it seems, the register definition is only for the H3 Pro model. So, it doesn't work for me.

**Reply by FozzieUK** [reply]
The latest protocol spec is only for the gen 2 inverters, the gen 1 inverters such as H1, H3 have very different register sets and MeterCompensation option has sadly never been exposed over modbus for the earlier generation inverters.
